profiler/main.py

```diff
@@ -0,0 +1,169 @@
+        peak_memory=peak_memory_footprint,
+        date=date.today(),
+        now=datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
+        tiledb_stats=tiledb_stats,
+        somacore_version=somacore.__version__,
+        tiledbsoma_version=tiledbsoma.__version__,
+    )
+    return data
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "process", nargs="+", help="The main process and its arguments to run"
+    )
+    parser.add_argument(
+        "-p1",
+        "--prof1",
+        required=False,
+        help="Second profiler and its args to use (like python and R) to produce flamegraph of calls",
+    )
+    parser.add_argument(
+        "-p1out",
+        "--prof1_output",
+        required=False,
+        help="The flamegraph output produced by prof1",
+    )
+
+    parser.add_argument(
+        "-p2",
+        "--prof2",
+        required=False,
+        help="Third profiler and its args to use (like python and R) to produce flamegraph of calls",
+    )
+    parser.add_argument(
+        "-p2out",
+        "--prof2_output",
+        required=False,
+        help="The flamegraph output produced by prof2",
+    )
+    args = parser.parse_args()
+
+    print(f"Process to be run: {args.process}")
+    # Running the main process using time -v to get detailed memory and time"""
+    p = subprocess.Popen(
+        ["/usr/bin/time", "-al"] + args.process, stdout=PIPE, stderr=PIPE
+    )
+    print(f"Running main process, PID = {p.pid}")
+
+    # Running additional profilers to extract flame graphs for the run
+    p1 = None
+    p2 = None
+    if args.prof1 is not None:
+        if args.prof1_output is not None:
+            command = args.prof1.replace("PID", str(p.pid))
+            p1 = subprocess.Popen(command.split())
+            # potential example of a custom profiler here is:
+            # p1 = subprocess.Popen(["py-spy", "record", "-o", "profile.svg", "--", "python", "tests/objects.py"])
+            print(f"Running second profiler {command.split()}")
+        else:
+            print(
+                f"Second profiler {args.prof1} missing output flamegraph file location"
+            )
+
+    if args.prof2 is not None:
+        if args.prof1_output is not None:
+            command = args.prof2.replace("PID", str(p.pid))
+            p2 = subprocess.Popen(command.split())
+            print(f"Running third profiler {command}")
+        else:
+            print(
+                f"Third profiler {args.prof2} missing output flamegraph file location"
+            )
+
+    stderr, stdout = p.communicate()
+    if p1 is not None:
+        p1.wait()
+    if p2 is not None:
+        p2.wait()
+
+    o: str = stderr.decode("utf-8")
+    print(f"The benchmarked process output:\n {o}")
+    # Parse the generated output from the time utility
+    data: ProfileData = parse_time_output(
+        stdout.decode("utf-8"), args.process, args.prof1_output, args.prof2_output
```

Comment:
nit: it's odd to pass in the `prof{1,2}_output` to a method called `parse...`.  maybe rename method to `build_profile_data`? or add in the extra output items after this call?